bundles/org.openhab.core.audio/src/main/java/org/openhab/core/audio/AudioFormat.java

```diff
@@ -436,29 +441,13 @@ public boolean isCompatible(@Nullable AudioFormat audioFormat) {
     @Override
     public boolean equals(@Nullable Object obj) {
         if (obj instanceof AudioFormat format) {
-            if (!(null == getCodec() ? null == format.getCodec() : getCodec().equals(format.getCodec()))) {
-                return false;
-            }
-            if (!(null == getContainer() ? null == format.getContainer()
-                    : getContainer().equals(format.getContainer()))) {
-                return false;
-            }
-            if (!(null == isBigEndian() ? null == format.isBigEndian() : isBigEndian().equals(format.isBigEndian()))) {
-                return false;
-            }
-            if (!(null == getBitDepth() ? null == format.getBitDepth() : getBitDepth().equals(format.getBitDepth()))) {
-                return false;
-            }
-            if (!(null == getBitRate() ? null == format.getBitRate() : getBitRate().equals(format.getBitRate()))) {
-                return false;
-            }
-            if (!(null == getFrequency() ? null == format.getFrequency()
-                    : getFrequency().equals(format.getFrequency()))) {
-                return false;
-            }
-            if (!(null == getChannels() ? null == format.getChannels() : getChannels().equals(format.getChannels()))) {
-                return false;
-            }
+            return Objects.equals(getCodec(), format.getCodec()) && //
```

Comment:
If all conditions are met, this now returns `true` while before it returned `super.equals(obj)`. Is this intended?

Reply:
Yes, I think that was a mistake, because if I understood it correctly that super.equals(obj) will execute the default implementation for object instances and return true only for the same instance, so the previous checks will always be true when true and having the methods overwrited makes no difference. I think the intention here was to consider equal formats with the same properties.